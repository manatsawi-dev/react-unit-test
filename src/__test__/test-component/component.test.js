import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Sinon from "sinon";
import View from "./component";

describe("<View/>", () => {
  it("renders View correctly", () => {
    const tree = renderer.create(<View />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders View append child correctly", () => {
    const tree = renderer
      .create(
        <View>
          <p>Child</p>
        </View>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders <View/> empty child correctly", () => {
    const wrapper = shallow(<View />);
    expect(wrapper.find("p").length).toEqual(0);
  });

  it("renders <View/> call function onClick correctly", () => {
    const onClickView = Sinon.spy();
    const wrapper = shallow(<View onClick={onClickView} />);
    wrapper.find("div").simulate("click");
    expect(onClickView).toHaveProperty("callCount", 1);
  });

  it("renders <View/> append child call function onClick at child correctly", () => {
    const onClickButton = Sinon.spy();
    const wrapper = shallow(
      <View>
        <button onClick={onClickButton}>Button</button>)
      </View>
    );
    wrapper.find("button").simulate("click");
    expect(onClickButton).toHaveProperty("callCount", 1);
  });
});
