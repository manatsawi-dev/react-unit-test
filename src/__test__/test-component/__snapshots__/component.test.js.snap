// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View/> renders View append child correctly 1`] = `
<div
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <p>
    Child
  </p>
</div>
`;

exports[`<View/> renders View correctly 1`] = `
<div
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;
